aea/helpers/async_utils.py

```diff
@@ -516,8 +516,9 @@ def start(self) -> bool:
 
         if self._threaded:
             self._thread = Thread(
-                target=self._loop.run_until_complete, args=[self._task]  # type: ignore # loop was set in set_loop
+                target=self._loop.run_until_complete, args=[self._task], name=self.__class__.__name__  # type: ignore # loop was set in set_loop
             )
+            self._thread.setDaemon(True)
```

Comment:
What consequences does this have for existing functionality?

Reply:
thread will be stopped on main thread exit, actually helps in case the developer forgets to join it.